projects/storefrontlib/cms-components/content/tab/tab.component.html

```diff
@@ -0,0 +1,66 @@
+<ng-container *ngIf="mode$ | async as mode">
+  <!-- Tab List is the list of buttons to open tabs -->
+  <div
+    [attr.role]="mode === TAB_MODE.ACCORDIAN ? 'presentation' : 'tablist'"
+    [attr.aria-label]="config.label | cxTranslate"
+    [class]="mode.toLowerCase()"
+  >
+    <ng-container *ngFor="let tab of tabs; let i = index">
+      <button
+        class="tab-btn"
+        [attr.role]="mode === TAB_MODE.ACCORDIAN ? 'button' : 'tab'"
+        (click)="select(i, mode)"
+        (keydown)="handleKeydownEvent(i, tabs, mode, $event)"
+        [attr.aria-selected]="mode === TAB_MODE.ACCORDIAN ? null : isOpen(i)"
+        [attr.aria-expanded]="mode === TAB_MODE.TAB ? null : isOpen(i)"
+        [attr.aria-controls]="'section-' + i"
+        [attr.tabindex]="getTabIndex(i, mode)"
+        [attr.id]="tab.id"
+        [attr.title]="
+          mode === TAB_MODE.TAB
+            ? null
+            : (isOpen(i) ? 'Collapse' : 'Expand') +
+              ' ' +
+              (tab.headerKey ? (tab.headerKey | cxTranslate) : tab.header)
```

Comment:
Complex logic inside the template, like calculating the title or tab indexes, can be moved to the component class to keep the template clean and more readable. what do you think?
example,

getTitle(i: number, mode: string): string {
  if (mode === TAB_MODE.TAB) {
    return null;
  }
  return (this.isOpen(i) ? 'Collapse' : 'Expand') + ' ' + (this.tabs[i].headerKey ? this.translate(this.tabs[i].headerKey) : this.tabs[i].header);
}